src/Nfts/ImageGallery/ImageGallery.tsx

```diff
@@ -18,34 +26,56 @@ type Props = {
 export const SkeletonGallery = ({amount}: {amount: number} = {amount: 3}) => {
   const placeholders = new Array(amount).fill(undefined)
   return (
-    <SafeAreaView edges={['left', 'right', 'bottom']} style={styles.safeAreaView}>
-      <ScrollView bounces={false} contentContainerStyle={styles.galleryContainer}>
-        {placeholders.map((item, index) => (
-          <SkeletonImagePlaceholder key={index} />
-        ))}
-      </ScrollView>
-    </SafeAreaView>
+    <ScrollView bounces={false} contentContainerStyle={styles.galleryContainer}>
+      {placeholders.map((item, index) => (
+        <SkeletonImagePlaceholder key={index} />
+      ))}
+    </ScrollView>
   )
 }
 
 export const ImageGallery = ({nfts = [], onSelect}: Props) => {
+  const rows = groupArray(nfts)
   return (
-    <SafeAreaView edges={['left', 'right', 'bottom']} style={styles.safeAreaView}>
-      <ScrollView bounces={false} contentContainerStyle={styles.galleryContainer}>
-        {nfts.map((nft, index) => {
-          const fingerprint = getAssetFingerprint(nft.metadata.policyId, nft.metadata.assetNameHex)
+    <ScrollView bounces={false} contentContainerStyle={styles.galleryContainer}>
+      <VirtualizedList
+        data={rows}
+        initialNumToRender={4}
+        getItem={(array, index) => array[index]}
+        getItemCount={() => rows.length}
+        keyExtractor={(row: Array<YoroiNFT | undefined>, index) =>
+          row[0] !== undefined
+            ? getAssetFingerprint(row[0].metadata.policyId, row[0].metadata.assetNameHex)
+            : String(index)
+        }
+        renderItem={({item: [nft1, nft2]}) => {
```

Comment:
is this intended to render 2 images per row? if so, FlatList takes a `numColumns` prop